test/controllers/pair_retro_controller_test.exs

```diff
@@ -0,0 +1,179 @@
+
+  @valid_attrs %{subject: "some content", reflection: "some content", pair_date: Timex.today}
+  @invalid_attrs %{}
+
+  test "redirects to sign-in when not logged in", %{conn: conn} do
+    conn = get conn, user_path(conn, :index)
+    assert redirected_to(conn) == session_path(conn, :new)
+  end
+
+  defp create_pair_and_retro(user) do
+    pair = create_pair([user])
+    retro = create_retro(user, pair)
+    {pair, retro}
+  end
+
+  defp create_user_and_pair_and_retro() do
+    user = insert(:user)
+    pair = create_pair([user])
+    retro = create_retro(user, pair)
+    {user, pair, retro}
+  end
+
+  describe "while authenticated" do
+    setup do
+      user = insert(:user)
+      conn = build_conn
+        |> log_in(user)
+      {:ok, [conn: conn, logged_in_user: user]}
+    end
+
+    test "lists all entries on index", %{conn: conn} do
+      conn = get conn, pair_retro_path(conn, :index)
+      assert html_response(conn, 200) =~ "Listing your retrospectives"
+    end
+
+    test "lists the current user's retrospectives", %{conn: conn, logged_in_user: user} do
+      {_pair, retro} = create_pair_and_retro(user)
+      conn = get conn, pair_retro_path(conn, :index)
+      assert html_response(conn, 200) =~ Ecto.Date.to_string(retro.pair_date)
+    end
+
+    test "does not list a retrospective of a different user", %{conn: conn} do
+      {_other_user, _pair, retro} = create_user_and_pair_and_retro
+      conn = get conn, pair_retro_path(conn, :index)
+      refute html_response(conn, 200) =~ Ecto.Date.to_string(retro.pair_date)
+    end
+
+    test "renders form for new resources", %{conn: conn, logged_in_user: user} do
+      pair = create_pair([user])
+      conn = get conn, pair_retro_path(conn, :new, pair.id)
+      assert html_response(conn, 200) =~ "New retrospective"
+    end
+
+    test "creates retro and redirects when data is valid and user is logged in", %{conn: conn, logged_in_user: user} do
+      pair = create_pair([user])
+      attrs = Map.merge(@valid_attrs, %{pair_id: Integer.to_string(pair.id),
+                                        user_id: Integer.to_string(user.id)})
+      conn = post conn, pair_retro_path(conn, :create), pair_retro: attrs
+      assert redirected_to(conn) == pair_retro_path(conn, :index)
+      assert Repo.get_by(PairRetro, attrs)
+    end
+
+    test "does not create retro with a pair_date before the pair's week", %{conn: conn, logged_in_user: user} do
+      pair = create_pair([user], 2016, 1)
+      attrs = Map.merge(@valid_attrs, %{pair_date: ~D(1999-10-20),
+                                        pair_id: Integer.to_string(pair.id),
+                                        user_id: Integer.to_string(user.id)})
+      conn = post conn, pair_retro_path(conn, :create), pair_retro: attrs
+      assert html_response(conn, 200) =~ "New retrospective"
+      assert html_response(conn, 200) =~ "cannot be before the week of the pair"
+    end
```

Comment:
The test around the behavior for creating a retrospective for a date that occurs in the future is contained in the `PairRetroTest` model tests, as all of its behaviors is contained in the changeset validation itself. This test exists in here, as the decision was made to pass in the date of the pair to the changeset. See the comment around the changeset for more details.